FirebaseAdmin/FirebaseAdmin/Messaging/InstanceIdClient.cs

```diff
@@ -145,24 +144,24 @@ private void ValidateRegistrationTokenList(List<string> registrationTokens)
         {
             if (registrationTokens == null)
             {
-                throw new FirebaseMessagingException(ErrorCode.InvalidArgument, "Registration token list must not be null");
+                throw new ArgumentNullException("Registration token list must not be null");
             }
 
             if (registrationTokens.Count() == 0)
             {
-                throw new FirebaseMessagingException(ErrorCode.InvalidArgument, "Registration token list must not be empty");
+                throw new ArgumentException("Registration token list must not be empty");
             }
 
             if (registrationTokens.Count() > 1000)
             {
-                throw new FirebaseMessagingException(ErrorCode.InvalidArgument, "Registration token list must not contain more than 1000 tokens");
+                throw new ArgumentException("Registration token list must not contain more than 1000 tokens");
             }
 
             foreach (var registrationToken in registrationTokens)
             {
                 if (string.IsNullOrEmpty(registrationToken))
                 {
-                    throw new FirebaseMessagingException(ErrorCode.InvalidArgument, "Registration token must not be null");
+                    throw new ArgumentNullException("Registration tokens must not be null");
```

Comment:
Throw `ArgumentException` and change the text to "Registration tokens must not be null or empty"